src/NHibernate/Multi/IQueryBatchItem.cs

```diff
@@ -32,43 +32,57 @@ public interface IQueryBatchItem<TResult> : IQueryBatchItem
 	public partial interface IQueryBatchItem
 	{
 		/// <summary>
-		/// The query loading information.
+		/// Optionally, the query caching information list, for batching. Each element matches
+		/// a SQL-Query resulting from the query translation, in the order they are translated.
+		/// It should be <see langword="null" /> if no batching of caching is handled for this
+		/// query.
 		/// </summary>
-		IEnumerable<IQueryInformation> QueryInformation { get; }
+		IEnumerable<ICachingInformation> CachingInformation { get; }
 
 		/// <summary>
-		/// Method is called right before batch execution.
+		/// Initialize the query. Method is called right before batch execution.
 		/// Can be used for various delayed initialization logic.
 		/// </summary>
 		/// <param name="session"></param>
 		void Init(ISessionImplementor session);
+
+		/// <summary>
+		/// Get the query spaces.
+		/// </summary>
+		/// <remarks>
+		/// Query spaces indicates which entity classes are used by the query and need to be flushed
+		/// when auto-flush is enabled. It also indicates which cache update timestamps needs to be
+		/// checked for up-to-date-ness.
+		/// </remarks>
+		IEnumerable<string> GetQuerySpaces();
 		
 		/// <summary>
-		/// Gets the commands to execute for getting the not-already cached results of this query.
+		/// Get the commands to execute for getting the not-already cached results of this query.
 		/// </summary>
 		/// <param name="cachedResults">The cached results.</param>
 		/// <returns>The commands for obtaining the results not already cached.</returns>
-		IEnumerable<ISqlCommand> GetCommands(IDictionary<IQueryInformation, IList> cachedResults);
+		IEnumerable<ISqlCommand> GetCommands(IDictionary<ICachingInformation, IList> cachedResults);
```

Comment:
It seems this change is not really required. Why can't we set all required information right in to `ICachingInformation`? 
You can add something like `ICachingInfromation.SetCachedResults` and call it right in `BatchItem.GetCachedResults`. So no additional lookup and less impact on base interface  for this specific feature. 

We can also provide reference to common `CacheBatcher`  via `ICachingInformation` - so it seems we also can get rid  of 'InitializeEntitiesAndCollections'. Thus making cache implementation exposed only via single property.